src/App.tsx

```diff
@@ -27,35 +34,101 @@ const useStyles = makeStyles((theme) => ({
   },
 }));
 
-const LanguageSelector = (props: LanguageSelectorProps): ReactElement => {
-  const [name, setName] = useState(props.name);
-  const classes = useStyles();
-  const tStyle = { color: 'white' };
+const enumStyles = makeStyles((theme) => ({
+  formControl: {
+    margin: theme.spacing(1),
+    minWidth: 100,
+  },
+  selectEmpty: {
+    marginTop: theme.spacing(2),
+  },
+}));
+
+const LanguageSelector = (): ReactElement => {
+  const [name, setName] = useState(supportedLanguages[langStore.getItem('lang') as string].name);
+  const classes = langStyles();
+  const textStyle = { color: "white" };
 
   const onValueChange = useCallback(
     (event: React.ChangeEvent<{ name?: string; value: unknown }>) => {
       const value = event.target.value as string;
-      setName(value);
+      langStore.setItem("lang", value);
+      setName(supportedLanguages[value].name);
+      window.location.reload();
     }, []
   );
 
+  const arrayMenuItems = [];
+  console.log(supportedLanguages);
+
+  for (let language in supportedLanguages) {
+    const languageName = supportedLanguages[language].name as string;
+    arrayMenuItems.push(<MenuItem value={language}>{languageName}</MenuItem>);
+  }
+
+
   return (
     <div>
       <Grid container>
         <FormControl className={classes.formControl}>
           <Select
             id="simple-menu"
             onChange={onValueChange}
-            defaultValue="English"
+            defaultValue={langStore.getItem('lang')}
             style={{ backgroundColor: "white" }}
```

Comment:
remove that, and assign a className instead.

Reply:
The classname could be just "whiteBackground", and you can assign it to all the components where you want them to have a white background for example.

Reply:
so here you can do className="whiteBgColor" and remove the style.